roles/monitor_agent_based_installer/tasks/main.yml

```diff
@@ -4,7 +4,31 @@
     chdir: "{{ manifests_dir }}"
   ignore_errors: True # Timeout is fixed and some bare metal clusters complete bootstrap just after the timeout
 
-- name: Wait for install complete
-  ansible.builtin.shell:
-    cmd: "{{ agent_based_installer_path }} --log-level=debug agent wait-for install-complete"
-    chdir: "{{ manifests_dir }}"
+- name: Check installation and gather jobs if it fails
+  block:
+    - name: Wait for install complete
+      ansible.builtin.shell:
+        cmd: "{{ agent_based_installer_path }} --log-level=debug agent wait-for install-complete"
+        chdir: "{{ manifests_dir }}"
+
+  rescue:
+    # Using master-0 IP address to reach the bootstrap VM
+    # Placing the logs in repo_root_path
+    # Trying several times in case there are SSH connectivity issues
+    - name: Gather logs from installation
+      vars:
+        bootstrap_host: >-
```

Comment:
The node should be selected like [this](https://github.com/redhat-partner-solutions/crucible/blob/35bebe48675ec73c0809338de4fa945bafc474bd/roles/generate_manifests/defaults/main.yml#L11) as that is how its specified as the [bootstrap IP](https://github.com/redhat-partner-solutions/crucible/blob/35bebe48675ec73c0809338de4fa945bafc474bd/roles/generate_manifests/templates/agent-config.yaml.j2#L4)

Reply:
Good to know it! Since it's defined in a different role, I'll create the same var for this task and just reuse it

Reply:
Yes just make sure you using the first master instead of the first node. They might be the same but they don't have to be :)